app/controllers/collections_controller.rb

```diff
@@ -121,6 +121,93 @@ def revoke_access
     redirect_to licences_path
   end
 
+  def add_items_to_collection
+    # referenced documents (HCSVLAB-1019) are already handled by the look_for_documents part of the item ingest
+    collection = Collection.find_by_name(params[:id])
+    if !request.post? or params[:items].blank?
+      respond_with_error("JSON-LD formatted item metadata must be sent to the api call as a POST request", 400)
+    elsif collection.nil?
+      respond_with_error("Requested collection not found", 404)
+    elsif params[:api_key] != User.find(collection.owner_id).authentication_token
+      respond_with_error("User is unauthorised", 403) # Authorise by comparing api key sent with collection owner's api key
```

Comment:
I think the authorisation is already handled by the Devise gem so you don't need to worry about checking that.